opwen_email_server/utils/email_parser.py

```diff
@@ -80,3 +83,28 @@ def _get_recipients(email: dict) -> Iterable[str]:
 def get_domains(email: dict) -> Iterable[str]:
     return frozenset(address.split('@')[-1]
                      for address in _get_recipients(email))
+
+
+def _get_as_base64(image_url):
+    res = requests.get(image_url)
+
+    if res.status_code != 200:
+        res.raise_for_status()
+
+    img_type = res.headers['Content-Type']
```

Comment:
This will crash if there is no content-type available on the response. Please handle that case gracefully, e.g. using `img_type = res.headers.get('Content-Type')`.

You can also use `guess_type` from the `mimetypes` package to try to guess the image type based on the filename. This is how we're currently setting the content type for email attachments in [sendgrid.py](https://github.com/ascoderu/opwen-cloudserver/blob/c3aaf804e8e0baf8cf65a9b02f39ddc5b0721b44/opwen_email_server/services/sendgrid.py#L114).